llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -581,6 +581,11 @@ static bool canProduceS20(const MachineRegisterInfo &MRI,
   case TargetOpcode::G_CONSTANT:
   case TargetOpcode::G_IMPLICIT_DEF:
     return true;
+  case TargetOpcode::G_INTRINSIC: {
+    const AIEBaseSubtarget &STI = AIEBaseSubtarget::get(*MI.getMF());
+    const AIEBaseInstrInfo *TII = STI.getInstrInfo();
+    return TII->isS20Intrinsic(MI, MRI);
+  }
```

Comment:
I'd rather have a pre-combiner than transforms constant-sign vextract intrinsics into `G_AIE_ZEXT_EXTRACT_VECTOR_ELT` or `G_AIE_SEXT_EXTRACT_VECTOR_ELT`. Then we do not need to define `isS20Intrinsic`.